src/components/contextual/pages/pool/PoolCompositionCard/components/composables/useTokenBreakdown.ts

```diff
@@ -1,66 +1,138 @@
-import { computed, Ref } from 'vue';
-import { bnum } from '@/lib/utils';
-import { PoolToken } from '@/services/pool/types';
 import useNumbers, { FNumFormats } from '@/composables/useNumbers';
+import { isDeep } from '@/composables/usePool';
+import { useUserPoolPercentage } from '@/composables/useUserPoolPercentage';
+import { bnum } from '@/lib/utils';
+import { isNumber } from '@/lib/utils/numbers';
+import { Pool, PoolToken } from '@/services/pool/types';
+import { computed, Ref } from 'vue';
+
+export interface TokenData {
+  balanceLabel: string | number;
+  userBalanceLabel: string | number;
+  fiatLabel: string;
+  userFiatLabel: string;
+  tokenWeightLabel: string;
+  getTokenPercentageLabel: () => string;
+}
 
-export function useTokenBreakdown(
-  token: Ref<PoolToken>,
-  shareOfParentInPool: Ref<number>,
-  isDeepPool: Ref<boolean>
-) {
+export type TokensData = Record<string, TokenData>;
+
+export function useTokenBreakdown(rootPool: Ref<Pool>) {
   const { fNum2, toFiat } = useNumbers();
 
-  // To get the balance of this token in the current pool we need to know the
-  // share of it's parent in this pool. e.g. If the token is DAI which is nested
-  // in bb-a-DAI, we need to know what share of bb-a-DAI is contained in the
-  // current pool. Then we can use this share and multiply it by the total
-  // balance of this token.
-  const balance = computed(() => {
-    return bnum(token.value.balance)
-      .times(shareOfParentInPool.value)
-      .toString();
-  });
-
-  const isParentTokenInDeepPool = computed(() => {
-    const hasNestedTokens = token.value?.token?.pool?.tokens;
-    return hasNestedTokens && isDeepPool.value;
-  });
-
-  const balanceLabel = computed(() => {
-    if (isParentTokenInDeepPool.value) return '';
-
-    if (token.value.priceRate && isDeepPool.value) {
-      const equivMainTokenBalance = bnum(balance.value)
-        .times(token.value.priceRate)
-        .toString();
-
-      return fNum2(equivMainTokenBalance, FNumFormats.token);
+  const { userPoolPercentage } = useUserPoolPercentage(rootPool);
+  const isDeepPool = computed(() => isDeep(rootPool.value));
+  let tokensData: TokensData = {};
+  // Sum of every token's fiat value.
+  // Used to increase token percentage accuracy as pool.totalLiquidity is not completely in sync with balance * coingecko-price calculations).
+  let totalFiat = 0;
+
+  // Recalculates recursive tokensData whenever the pool is re-fetched
+  return computed(() => calculateAllTokensData(rootPool));
+
+  function calculateAllTokensData(pool: Ref<Pool>) {
```

Comment:
This code is good, but I think it could be moved to the SDK (later) as it would be useful to have a `getUserBalances` function on a pool that takes a user address and returns all the details about the investment they have in the pool. @gmbronco

Reply:
> LGTM - just the one issue with veBAL pool (not sure if this is fixable becuase the BPT are locked?) and that a lot of the calculation code should be in the SDK instead of frontend.

It is interesting that it does not show the tab because I use this query:

`  const { stakedShares } = usePoolStaking();
`

https://github.com/balancer-labs/frontend-v2/pull/2709/files#diff-9dd467c6218b8ac7397d3cee3fcddca3e89a1faacceb00ac07e0a4fd6a749075R10

That it's returning 0 for `timjrobinson.eth` in this pool (http://localhost:8080/#/ethereum/pool/0x5c6ee304399dbdb9c8ef030ab642b10820db8f56000200000000000000000014)

@timjrobinson or @garethfuller Do you know why?

Reply:
For this specific pool you don't stake your shares, you lock them, so they don't feature in `stakedShares`. You need to use `useLock`. And the value you're looking for is `lock.value.lockedAmount` https://github.com/balancer-labs/frontend-v2/blob/develop/src/composables/useLock.ts#L53.

Reply:
Ok, that makes sense. I think that we could merge this current branch and creating an independent task for this (Showing user locked shares in composition table).